gui/slick/js/ajax-episode-subtitles.js

```diff
@@ -1,59 +1,189 @@
+        });
+    }
+
+    function enableAllSearches() {
+        $.each(subtitleTypeList, function (index, subtitleType) {
+            $(subtitleType).css({'pointer-events' : 'auto'});
+        });
+    }
+
+    function changeImage(imageTR, srcData, altData, titleData, heightData, emptyLink) {
+        if (emptyLink === true) {
+            imageTR.empty();
+        }
+        imageTR.append($('<img/>').prop({
+            src: srcData,
+            alt: altData,
+            title: titleData,
+            width: 16,
+            height: heightData
+        }));
+    }
+
+    $('#manualSubtitleSearchModal').on('hidden.bs.modal', function () {
+        // If user close manual subtitle search modal, enable again all searches
+        enableAllSearches();
+    });
+
     $.ajaxEpSubtitlesSearch = function() {
-        $('.epSubtitlesSearch').on('click', function() {
-            var subtitlesTd = $(this).parent().siblings('.col-subtitles');
-            var subtitlesSearchLink = $(this);
-            // fill with the ajax loading gif
-            subtitlesSearchLink.empty();
-            subtitlesSearchLink.append($('<img/>').prop({
-                src: 'images/loading16.gif',
-                alt: '',
-                title: 'loading'
-            }));
-            $.getJSON($(this).attr('href'), function(data) {
+        $('.epSubtitlesSearch').on('click', function(e) {
+            // This is for the page 'displayShow.mako'
+            e.preventDefault();
+            selectedEpisode = $(this);
+            subtitlesTd = selectedEpisode.parent().siblings('.col-subtitles');
+            // Ask user if he want to manual search subs or automatic search
+            $('#askmanualSubtitleSearchModal').modal('show');
+        });
+
+        $('.epSubtitlesSearchPP').on('click', function(e) {
+            // This is for the page 'manage_subtitleMissedPP.mako'
+            e.preventDefault();
+            selectedEpisode = $(this);
+            subtitlesTd = selectedEpisode.parent().siblings('.col-search');
+            searchSubtitles();
+        });
+
+        $(document).on('click', '#pickSub', function(event){
+            subtitlePicked = $(this);
+            changeImage(subtitlePicked, 'images/loading16.gif', 'loading', 'loading', 16, true);
+            var subtitleID = subtitlePicked.attr('subtitleID');
+            // Remove 'subtitleid-' so we know the actual ID
+            subtitleID = subtitleID.replace('subtitleid-', '');
+            url = selectedEpisode.prop('href');
+            url = url.replace('searchEpisodeSubtitles', 'manual_search_subtitles');
+            // Append the ID param that 'manual_search_subtitles' expect when picking subtitles
+            url += '&picked_id=' + subtitleID;
+            $.getJSON(url, function(data) {
+                // If user click to close the window before subtitle download finishes, show again the modal
+                if (($('#manualSubtitleSearchModal').is(':visible')) === false) {
```

Comment:
The `#manualSubtitleSearchModal` selector should be cached to save memory usage. This will be added to the xo linter soon as all new code should be following this.

Just set a var that's equal to the jQuery selector and use the var instead.